src/main/java/com/jcabi/log/SupplierLogger.java

```diff
@@ -0,0 +1,109 @@
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.log;
+
+/**
+ * Logging methods which take {@link Supplier} arguments.
+ * Used with Java 8 method referencing.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ * @since 0.18
+ */
+final class SupplierLogger {
+
+    /**
+     * Log one message, with {@code TRACE} priority level.
+     * @param source The source of the logging operation
+     * @param msg The text message to be logged, with meta-tags
+     * @param args List of {@link Supplier} arguments. Method
+     *  references will be mapped to Supplier instances.
+     */
+    public void trace(
+        final Object source, final String msg, final Supplier<?>... args) {
+        if (Logger.isTraceEnabled(source)) {
+            Logger.traceForced(source, msg, this.supplied(args));
+        }
+    }
+
+    /**
+     * Log one message, with {@code DEBUG} priority level.
+     * @param source The source of the logging operation
+     * @param msg The text message to be logged, with meta-tags
+     * @param args List of {@link Supplier} arguments. Method
+     *  references will be mapped to Supplier instances.
+     */
+    public void debug(
+        final Object source, final String msg, final Supplier<?>... args) {
+        if (Logger.isDebugEnabled(source)) {
+            Logger.debugForced(source, msg, this.supplied(args));
+        }
+    }
+
+    /**
+     * Log one message, with {@code INFO} priority level.
+     * @param source The source of the logging operation
+     * @param msg The text message to be logged, with meta-tags
+     * @param args List of {@link Supplier} arguments. Method
+     *  references will be mapped to Supplier instances.
+     */
+    public void info(
+        final Object source, final String msg, final Supplier<?>... args) {
+        if (Logger.isInfoEnabled(source)) {
+            Logger.infoForced(source, msg, this.supplied(args));
+        }
+    }
+
+    /**
+     * Log one message, with {@code WARN} priority level.
+     * @param source The source of the logging operation
+     * @param msg The text message to be logged, with meta-tags
+     * @param args List of {@link Supplier} arguments. Method
+     *  references will be mapped to Supplier instances.
+     */
+    public void warn(
+        final Object source, final String msg, final Supplier<?>... args) {
+        if (Logger.isWarnEnabled(source)) {
+            Logger.warnForced(source, msg, this.supplied(args));
+        }
+    }
+
```

Comment:
@amihaiemil maybe I'm missing something, but if you say so why there's method http://www.slf4j.org/api/org/slf4j/Logger.html#isErrorEnabled()?